### PR TITLE
Add code for the new t2t genome

### PR DESCRIPTION
**Pull request by cadley-nyulangone**

### CODE REVIEW

File: dnase/trackhub/update_tracks.bash [hunk 1]
```diff
@@ -294,12 +302,19 @@ else
...
 fi
 
-if [ ${blatport} -ne 0 ]; then
+if [ "${blatport}" != "0" ]; then
     echo "Copying ${customGenomeAssembly}.2bit to cadlej01_shared..."
     cp -p ${hub_target_final}/${customGenomeAssembly}/data/${customGenomeAssembly}.2bit /vol/isg/cadlej01_shared
-    
+    twoBitFiles="${customGenomeAssembly}.2bit"
+
+    if [ "${blatport}" = "17779" ]; then
```
Comment (mattmaurano) [inline dnase/trackhub/update_tracks.bash:310]: what's this testing?

File: dnase/trackhub/update_tracks.bash [hunk 2]
```diff
@@ -213,6 +213,7 @@ cp -R assets/${hub_type}/. ${hub_target}
 
 for i in "${genome_array[@]}"; do
```
Comment (mattmaurano) [inline dnase/trackhub/update_tracks.bash:214]: what does this loop do?

Comment (mattmaurano) [inline dnase/trackhub/update_tracks.bash:214]: is i an integer?

File: dnase/trackhub/update_tracks.bash [hunk 3]
```diff
@@ -213,6 +213,7 @@ cp -R assets/${hub_type}/. ${hub_target}
...
     [ "${i}" = "${customGenomeAssembly}" ] && continue
+    [ "${i}" = "t2t" ] && continue
```
Comment (mattmaurano) [inline dnase/trackhub/update_tracks.bash:216]: why skipping these?

File: dnase/trackhub/MakeTrackhub.py [hunk 4]
```diff
@@ -163,6 +163,8 @@ def shortest_unique_strings(array, minlength=1):
         supertrackonoff="show")
     if args.genome == "cegsvectors":
         supertrack.add_params(group="cegsvectors")
+    if args.genome == "t2t":
+        supertrack.add_params(group="t2t")
```
Comment (mattmaurano) [inline dnase/trackhub/MakeTrackhub.py:167]: combine these two by doing:
supertrack.add_params(group=args.genome)

File: dnase/trackhub/update_tracks.bash [hunk 5]
```diff
@@ -294,12 +302,19 @@ else
     blatport=0
 fi
 
-if [ ${blatport} -ne 0 ]; then
+if [ "${blatport}" != "0" ]; then
```
Comment (mattmaurano) [inline dnase/trackhub/update_tracks.bash:305]: is this a stylistic or functional change?

**Review by mattmaurano — CHANGES_REQUESTED** [review]

File: dnase/trackhub/update_tracks.bash (hunk 1 above)
Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:310]: 17779 is the port number assigned to the blat server for the CEGS hub.  If we are building a hub for that port, we want to add t2t as well.

File: dnase/trackhub/update_tracks.bash (hunk 2 above)
Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:214]: This loop is building the genomes.txt file for the hub.  The members of genome_array are the various genomes supported by the hub, like hg38 and t2t.

Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:214]: No.  i is hg38, mm10, rn6, etc.

File: dnase/trackhub/update_tracks.bash (hunk 3 above)
Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:216]: The genomes native to the UCSC browser have simple, standard, two line stanzas in the genomes.txt file. They are written to genomes,txt here. The custom assemblies have more complicated, non-standard stanzas.  Those stanza lines are stored in the stub genomes.txt file in the assets/CEGS subdirectory, so we skip building them in this loop.

File: dnase/trackhub/MakeTrackhub.py (hunk 4 above)
Comment (cadley-nyulangone) [reply dnase/trackhub/MakeTrackhub.py:167]: OK

File: dnase/trackhub/update_tracks.bash (hunk 5 above)
Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:305]: Just stylistic.

File: dnase/trackhub/update_tracks.bash (hunk 1 above)
Comment (mattmaurano) [reply dnase/trackhub/update_tracks.bash:310]: what other blat servers are there?

File: dnase/trackhub/update_tracks.bash (hunk 2 above)
Comment (mattmaurano) [reply dnase/trackhub/update_tracks.bash:214]: then rename to something more descriptive like cur_genome

use i only for integers

File: dnase/trackhub/update_tracks.bash (hunk 3 above)
Comment (mattmaurano) [reply dnase/trackhub/update_tracks.bash:216]: add this info directly as a comment, also better to combine the two into a single if statement

File: dnase/trackhub/update_tracks.bash (hunk 1 above)
Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:310]: 17778 for MAURANOLAB.

File: dnase/trackhub/update_tracks.bash (hunk 2 above)
Comment (mattmaurano) [reply dnase/trackhub/update_tracks.bash:214]: add directly as a comment

File: dnase/trackhub/update_tracks.bash (hunk 1 above)
Comment (mattmaurano) [reply dnase/trackhub/update_tracks.bash:314]: don't key off of secondary features. check this way instead:
if [[ "${hub_type}" == "CEGS" ]]; then

File: dnase/trackhub/update_tracks.bash [hunk 6]
```diff
@@ -211,16 +211,27 @@ cd ${src}
...
+    echo "trackDb ${cur_genome}/trackDb_001.txt" >> "${hub_target}/genomes.txt"
     echo " " >> "${hub_target}/genomes.txt"
 done
 
 if [ -f "${assemblyBaseDir}/sequences/${customGenomeAssembly}/${customGenomeAssembly}.2bit" ]; then
     cp ${assemblyBaseDir}/sequences/${customGenomeAssembly}/${customGenomeAssembly}.2bit ${hub_target}/${customGenomeAssembly}/data/${customGenomeAssembly}.2bit
 fi
+
+if [[ "${hub_type}" == "CEGS" ]]; then
+    if [ ! -d "${hub_target}/t2t/data" ]; then
```
Comment (mattmaurano) [inline dnase/trackhub/update_tracks.bash:230]: one last question, how could this directory have been created before?

Comment (cadley-nyulangone) [reply dnase/trackhub/update_tracks.bash:230]: Was just checking.  It is not though.  I'll take the if out.

**Pull request merged**